File: CSharpOddOrEven/CSharpOddOrEven/OddOrEven.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.Logging;
using System.Numerics;

namespace CSharpOddOrEven
{
    public static class OddOrEven
    {
        [FunctionName("OddOrEven")]
        public static IActionResult Run([HttpTrigger(AuthorizationLevel.Function, "get", "post")]HttpRequest req, ILogger log)
        {
            log.LogInformation("Odd or even trigger fired - HTTP");

            string numberQueryValue = req.Query["number"];

            if (BigInteger.TryParse(numberQueryValue, out BigInteger number))
            {
                return new OkObjectResult(number % 2 == 0 ? "Even" : "Odd");
            }
            else
            {
                return new BadRequestObjectResult($"Unable to parse the query parameter 'number'. Got value: {numberQueryValue}");
            }
        }
    }
}

File: JavaOddOrEven/src/main/java/com/function/OddOrEven.java
package com.function;

import java.util.*;
import com.microsoft.azure.functions.annotation.*;
import com.microsoft.azure.functions.*;

/**
 * Azure Functions with HTTP Trigger.
 */
public class OddOrEven {

    @FunctionName("HttpTrigger-Java")
    public HttpResponseMessage HttpTriggerJava(
            @HttpTrigger(name = "req", methods = {HttpMethod.GET, HttpMethod.POST}, authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context) {
        try {
            context.getLogger().info("Java HTTP trigger processed a request.");
        
            String numberQueryValue = request.getQueryParameters().get("number");
            int number = Integer.parseInt(numberQueryValue);

            return request
                .createResponseBuilder(HttpStatus.OK)
                .body(number % 2 == 0 ? "Even" : "Odd")
                .build();
        } 
        catch(NumberFormatException nfe) {
            return request
                .createResponseBuilder(HttpStatus.BAD_REQUEST)
                .body("Unable to parse")
                .build();
        }
    }
}
